packages/universal-router-sdk/src/utils/routerCommands.ts

```diff
@@ -0,0 +1,243 @@
+  [CommandType.OWNER_CHECK_1155]: parseAbiParameters('address owner, address token, uint256 id, uint256 minBalance'),
+  [CommandType.APPROVE_ERC20]: parseAbiParameters('address token, uint256 spender'),
+
+  // NFT Markets
+  [CommandType.SEAPORT_V1_5]: parseAbiParameters('uint256 value, bytes data'),
+  [CommandType.SEAPORT_V1_4]: parseAbiParameters('uint256 value, bytes data'),
+  // @fixme: contract not implemented
+  [CommandType.LOOKS_RARE_V2]: parseAbiParameters('uint256 value, bytes data'),
+  [CommandType.X2Y2_721]: parseAbiParameters('uint256 value, bytes data, address recipient, address token, uint256 id'),
+  [CommandType.X2Y2_1155]: parseAbiParameters(
+    'uint256 value, bytes data, address recipient, address token, uint256 id, uint256 amount',
+  ),
+  [CommandType.PANCAKE_NFT_WBNB]: parseAbiParameters('address collection, uint256 tokenId, uint256 price'),
+  [CommandType.PANCAKE_NFT_BNB]: parseAbiParameters('address collection, uint256 tokenId, uint256 price'),
+  // @notice: following marketplace not supported now
+  // [CommandType.NFTX]: parseAbiParameters('uint256 value, bytes data'),
+  // [CommandType.FOUNDATION]: parseAbiParameters(
+  //   'uint256 value, bytes data, address recipient, address token, uint256 id'
+  // ),
+  // [CommandType.SUDOSWAP]: parseAbiParameters('uint256 value, bytes data'),
+  // [CommandType.NFT20]: parseAbiParameters('uint256 value, bytes data'),
+  // [CommandType.CRYPTOPUNKS]: parseAbiParameters('uint256 punkId, address recipient, uint256 value'),
+  // [CommandType.ELEMENT_MARKET]: parseAbiParameters('uint256 value, bytes data'),
+}
+
+export type CommandUsed = keyof typeof ABI_PARAMETER
+
+export class RoutePlanner {
+  commands: Hex
+
+  inputs: Hex[]
+
+  constructor() {
+    this.commands = '0x'
+    this.inputs = []
+  }
+
+  addSubPlan(subplan: RoutePlanner): void {
+    this.addCommand(CommandType.EXECUTE_SUB_PLAN, [subplan.commands, subplan.inputs], true)
+  }
+
+  addCommand<TCommandType extends CommandUsed>(
+    type: TCommandType,
+    parameters: ABIParametersType<TCommandType>,
+    allowRevert = false,
+  ): void {
+    const command = createCommand(type, parameters)
+    this.inputs.push(command.encodedInput)
+    if (allowRevert) {
+      if (!REVERTIBLE_COMMANDS.has(command.type)) {
+        throw new Error(`command type: ${command.type} cannot be allowed to revert`)
+      }
+      command.type |= ALLOW_REVERT_FLAG
+    }
+
+    this.commands = this.commands.concat(command.type.toString(16).padStart(2, '0')) as Hex
+  }
+}
+
+export type RouterCommand = {
+  type: CommandUsed
+  encodedInput: Hex
+}
+
+export function createCommand<TCommandType extends CommandUsed>(
+  type: TCommandType,
+  parameters: ABIParametersType<TCommandType>,
+): RouterCommand {
+  // const params = parameters.filter((param) => param !== null)
+  const encodedInput = encodeAbiParameters(ABI_PARAMETER[type], parameters as any)
```

Comment:
is it possible to remove the any type assertion?

Reply:
not sure wil ask jerry on this

Reply:
cc @ChefJerry

Reply:
maybe not, leave it `as any`, have tried but hard